interfaces/builtin/adb_support.go

```diff
@@ -0,0 +1,205 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const adbSupportSummary = `allows access to android devices for fastboot and adb`
+
+const adbSupportBaseDeclarationSlots = `
+  adb-support:
+    allow-installation:
+      slot-snap-type:
+        - core
```

Comment:
If we go with the slot implementation approach, then this would be:
```
  adb:
    allow-installation:
      slot-snap-type:
        - app
        - core
    deny-auto-connection: true
    deny-connection:
      on-classic: false
```

if we go with the slot implementation without implicit classic, this becomes:
```
  adb:
    allow-installation:
      slot-snap-type:
        - app
    deny-auto-connection: true
    deny-connection: true
```

Reply:
I'm picking the first approach as I think it would make sense for ubuports (since most of their audience will run classic systems).

Reply:
I'm not exactly sure what "implicit classic" means. The UBports Installer usually runs in strict confinement. Or is a classic system just a normal desktop distribution where not all apps are confined? Then yes, that is correct.

Reply:
@NeoTheThird - 'classic' is an overloaded term. In this sense, it means 'classic distro', aka, it is not an 'all-snaps' system like Ubuntu Core.

Reply:
Ok. So in that sense the vast majority of users of the UBports Installer (and other adb applications i can think of) will be on classic distro at the moment.